catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/QueryMonitorPluginImpl.java

```diff
@@ -0,0 +1,170 @@
+     * and {@link QueryRequest} information to create a new {@link ActiveSearch} to add to the ActiveSearchTable.
+     *
+     * @param source {@link Source} that corresponds to source the search is querying
+     * @param input  {@link QueryRequest} that corresponds to request generated when a user queried the source
+     * @return {@link QueryRequest} that was given as a parameter with updated property information corresponding
+     * to the {@link ActiveSearch}'s {@link UUID}
+     */
+
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        try {
+            if (source == null) {
+                LOGGER.warn("Source given is null.");
+            }
+            UUID uniqueID = UUID.randomUUID();
+            String cqlQuery = null;
+            if (input == null) {
+                LOGGER.warn("QueryRequest given to queryMonitor was null!");
+            } else {
+
+                cqlQuery = input.toString();
+                if (cqlQuery.equals("unknownCQLQuery")) {
+                    LOGGER.warn("CQL Query not set as expected in QueryRequest");
+                } else {
+                    int filterStartIndex = cqlQuery.indexOf("filter=[");
+                    int endOfQueryIndex = cqlQuery.indexOf(" ],startIndex");
+                    if (filterStartIndex == -1 || endOfQueryIndex == -1) {
+                        LOGGER.warn(
+                                "Unable to determine start and end of query terms. Using entire queryRequest instead.");
+                    } else {
+                        cqlQuery = cqlQuery.substring(filterStartIndex + 9, endOfQueryIndex);
+                    }
+                }
+                input.getProperties()
+                        .put("GUID", uniqueID);
+            }
+            addActiveSearch(new ActiveSearch(cqlQuery, source, uniqueID));
+        } catch (Exception e) {
+            LOGGER.warn(
+                    "Exception in queryPlugin monitor. Table may not contain entry for search that caused the exception. Query will continue past queryMonitorPlugin as usual.");
+        }
+        return input;
+    }
+
+    /**
+     * Method that is implemented for {@link PostFederatedQueryPlugin}. Uses the given {@link QueryResponse} information
+     * to remove the {@link ActiveSearch} from the ActiveSearchTable.
+     *
+     * @param input {@link QueryResponse} that corresponds to response from the source that was queried
+     *                                   by the user's original {@link QueryRequest}
+     * @return {@link QueryResponse} that was given as a parameter
+     */
+    @Override
+    public QueryResponse process(QueryResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        if (!removeFromTable) {
+            LOGGER.warn(
```

Comment:
This probably shouldn't be logged at a `warn` level. `debug` would probably be more appropriate.